gpusim.cpp

```diff
@@ -136,6 +175,8 @@ void GPUSimServer::extractData(const QString& database_fname,
                                 vector<char*>& smiles_vector,
                                 vector<char*>& ids_vector)
 {
+    vector<vector<char*> > smiles_data;
+    vector<vector<char*> > ids_data;
```

Comment:
declare these where you use them, so down on 214 and 225

Reply:
Also, why not have each QRunnable hold onto its own vector, and then iterate over each to insert into smiles_vector/ids_vector?

Reply:
Comment 2 answered above.  I'll move these down.